drivers/i2c/i2c_esp32.c

```diff
@@ -137,17 +132,15 @@ static int i2c_esp32_config_pin(const struct device *dev)
 		return -EINVAL;
 	}
 
-	gpio_pin_set(data->sda_gpio, config->sda.pin, 1);
-	ret = gpio_pin_configure(data->sda_gpio, config->sda.pin,
-			GPIO_PULL_UP | GPIO_OPEN_DRAIN | GPIO_OUTPUT | GPIO_INPUT);
-	esp_rom_gpio_matrix_out(config->sda.pin, config->sda.sig_out, 0, 0);
-	esp_rom_gpio_matrix_in(config->sda.pin, config->sda.sig_in, 0);
+	gpio_pin_set_dt(&config->sda.gpio, 1);
+	ret = gpio_pin_configure_dt(&config->sda.gpio, GPIO_PULL_UP | GPIO_OUTPUT | GPIO_INPUT);
```

Comment:
Shouldn't the `GPIO_PULL_UP` flag be in DTS instead? A board could have external pull-up resistors.

Reply:
I wasn't sure about that since, there is code that also configure's the gpio w/o `GPIO_PULL_UP` in `i2c_master_clear_bus`.

Reply:
@sylvioalves any comments on the use of `GPIO_PULL_UP` and why it might not be there for `i2c_master_clear_bus`

Reply:
We need to move all `GPIO_PULL_UP` hardcoded instances to DTS instead. `i2c_master_clear_bus` calls should not change `GPIO_PULL_UP` mode, only input/output/od modes. We can do this in another PR.

Reply:
> We need to move all `GPIO_PULL_UP` hardcoded instances to DTS instead. `i2c_master_clear_bus` calls should not change `GPIO_PULL_UP` mode, only input/output/od modes. We can do this in another PR.

If `i2c_master_clear_bus` should NOT be setting `GPIO_PULL_UP` than I think its better to leave the code as is and not change the DTS flags from what I've got here.